### ISSUE TITLE
ava 4.0.1 does not work with non-native ESM, i.e., esm shim

### ISSUE THREAD

**Issue by tcollinsworth** [avajs/ava#2951]
@tcollinsworth AVA 4 works just fine with CJS. Being a Node.js test runner, for ESM support we rely on Node.js' standard behavior.

_Originally posted by @novemberborn in https://github.com/avajs/ava/issues/2293#issuecomment-1019308184_

@novemberborn  Upgrading ava from 3.15.0 to 4.0.1 with esm-wallaby shim for optional-chaining support throws:
```
SyntaxError: Cannot use import statement outside a module
```

**Comment by novemberborn** [comment]
I'm not familiar with that shim. AVA 3 has special handling for the esm shim and that has been removed. You'll have to register it as a CJS loader somehow.

**Comment by tcollinsworth** [comment]
It's just a fork of the esm module but updated to support optional chaining (elvis operator). It works in ava 3.15.0, but fails with that error when updated to ava 4.0.1.

**Comment by novemberborn** [comment]
Correct. We used to recognize a symbol from the `esm` package and use it for subsequent requires, but that has been removed. I imagine there's a way to make it work like say `@babel/register` but I haven't looked closely.

**Comment by novemberborn** [comment]
You can specify node arguments to turn on experimental Node.js features. AVA does not do that by itself.
